collect_logs.py

```diff
@@ -0,0 +1,164 @@
+        ssh_install.exec_command(f"mv {log_dir} tmp/cephci-jenkins/ceph_logs")
+    except Exception:
+        results[nodeip] = False
+
+
+def collect_logs(nodeip, ssh_install, uname, log_dir, results):
+    """
+    Collect log from installer node based on ceph version
+    :param:
+       - nodeip: installer node ip
+       - ssh_install: ssh_install: ssh object of installer node
+       - uname: Username of installer node
+       - log_dir: directory to store all the logs
+       - results: dict: host Ip address with boolean parameters
+    """
+    try:
+        stdin, stdout, stderr = ssh_install.exec_command("sudo ceph --version")
+        if not stderr.read().decode():
+            file_name = "cephansible.sh"
+        else:
+            file_name = "cephadm.sh"
+        file = os.path.join(os.path.dirname(__file__), f"{file_name}")
+        sftp = ssh_install.open_sftp()
+        sftp.put(f"{file}", f"/home/{uname}/{file}")
+        ssh_install.exec_command(
+            f"sudo sh {file} > ceph_status.txt; sudo mv ceph_status.txt {log_dir}"
+        )
+    except Exception:
+        results[nodeip] = False
+
+
+def generate_sosreport_in_node(ssh_install, nodeip, uname, pword, log_dir, results):
+    """
+    Generate sosreport in the given node and copy report to installer
+    :param:
+       - ssh_install: ssh object of installer node
+       - nodeip: host Ip address
+       - uname: Username
+       - pword: password
+       - log_dir: directory to store all the logs
+       - results: dict: host Ip address with boolean parameters
+    """
+    print(f"Connecting {nodeip} to generate sosreport")
+    try:
+        results[nodeip] = True
+        ssh_d = SSHConnectionManager(nodeip, uname, pword).get_client()
+        ssh_d.exec_command("sudo yum -y install sos")
+        stdin, stdout, stderr = ssh_d.exec_command(
+            "sudo sosreport -a --all-logs -e ceph --batch"
+        )
+        sosreport = re.search(r"/var/tmp/sosreport-.*.tar.xz", stdout.read().decode())
+        print(f"Successfully generated sosreport in node {nodeip} :{sosreport.group()}")
+        ssh_d.exec_command(f"sudo chown {uname}:{uname} {sosreport.group()}")
+        ssh_install.exec_command(f"scp {nodeip}:{sosreport.group()} {log_dir}")
+        ssh_d.exec_command(f"sudo rm -rf {sosreport.group()}")
+        ssh_d.close()
+        print(f"Successfully copied report to {nodeip}")
+    except Exception:
+        results[nodeip] = False
+
+
+def run(args):
+    """
+    Using the provided installer ip get all other nodes then generate sosreport for all the nodes obtained.
+    Run shell script on installer node based on its ceph version, then upload all the collected logs to magna
+    :param:
+       - ip        :installer IP address
+       - username  :Username other than root
+       - password  :password for installer node
+       - log-dir   :name of the output directory [default tmp]
```

Comment:
IMO, we could do the docString like below... it helps render the documentation properly with readthedocs
```
Args:
  ip                  Node IP address or FQDN
  username     the username to be used to access the system
  log-dir           the place to store the collected logs
```
`log-dir`, I thought we were planning to use `magna002`